internal/pkg/plugin/githubactions/golang/validate.go

```diff
@@ -0,0 +1,73 @@
+package golang
+
+import (
+	"fmt"
+
+	"github.com/merico-dev/stream/pkg/util/log"
+	"github.com/mitchellh/mapstructure"
+)
+
+// validate validates the options provided by the core.
+func validate(opts *Options) []error {
+	retErrors := make([]error, 0)
+
+	// owner/repo/branch
+	if opts.Owner == "" {
+		retErrors = append(retErrors, fmt.Errorf("owner is empty"))
+	}
+	if opts.Repo == "" {
+		retErrors = append(retErrors, fmt.Errorf("repo is empty"))
+	}
+	if opts.Branch == "" {
+		retErrors = append(retErrors, fmt.Errorf("branch is empty"))
+	}
+
+	// language
+	if opts.Language == nil {
+		retErrors = append(retErrors, fmt.Errorf("language is empty"))
+	}
+	if errs := opts.Language.Validate(); len(errs) != 0 {
+		for _, e := range errs {
+			retErrors = append(retErrors, fmt.Errorf("language is invalid: %s", e))
+		}
+	}
+
+	// jobs
+	if opts.Test == nil {
+		retErrors = append(retErrors, fmt.Errorf("test is empty"))
+	}
+	if errs := opts.Test.Validate(); len(errs) != 0 {
+		for _, e := range errs {
+			retErrors = append(retErrors, fmt.Errorf("test is invalid: %s", e))
+		}
+	}
+
+	if opts.Docker == nil {
+		return retErrors
+	}
+
+	if errs := opts.Docker.Validate(); len(errs) != 0 {
+		for _, e := range errs {
+			retErrors = append(retErrors, fmt.Errorf("docker is invalid: %s", e))
+		}
+	}
+
+	return retErrors
+}
+
+func parseAndValidateOptions(options map[string]interface{}) (*Options, error) {
```

Comment:
This function should not exist because it makes the entry of the corresponding plugins inconsistent with the template (including most other plugins).

- template/create.go
 
```go
func Create(options map[string]interface{}) (map[string]interface{}, error) {
	var opts Options
	if err := mapstructure.Decode(options, &opts); err != nil {
		return nil, err
	}

	if errs := validate(&opts); len(errs) != 0 {
		for _, e := range errs {
			log.Errorf("Options error: %s.", e)
		}
		return nil, fmt.Errorf("opts are illegal")
	}

    // TODO(dtm): Add your logic here.

    return nil, nil
}
```

/cc @warren830 @summingyu Please also pay attention to this tip.